### PR TITLE
Creating a mailadm bot as an alternative way to give commands

### PR DESCRIPTION
**Pull request by missytake**
`mailadm setup-bot` seems to work so far; it completes in a docker container. The next step would be to test it on a real server.

Unfortunately, we can't easily test the whole setup in docker. There is an error during `./install_mailadm.sh`: https://github.com/alehaa/docker-debian-systemd/issues/3

- [ ] test on a real server

After this is merged, we can work on making more commands accessible through Delta Chat.

### CODE REVIEW

File: src/mailadm/db.py [hunk 1]
```diff
@@ -8,6 +8,16 @@
...
 
+def read_connection():
+    mailadm_db = DB(get_db_path())
+    return mailadm_db.read_connection()
+
+
+def write_connection():
+    mailadm_db = DB(get_db_path())
+    return mailadm_db.write_transaction()
+
```
Comment (hpk42) [inline src/mailadm/db.py:19]: These functions implicitely get the database path from global state (an environment variable).  Previously precise references were passed around.  I can see how you think it's easier to have global state but i consider it rather questionable (or even bad) practise.

**Review by hpk42 — CHANGES_REQUESTED** [review]
As you said yourself the PR mixes several things apart from adding a bot.  Moreover, the refactoring to make creatiing a DB object dependent on an env variable ... i'd rather avoid that.  
Moreover, i am not sure this way of adding bot support is a good idea ... Maybe we can do a little call about it.

File: src/mailadm/bot.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+from deltachat import account_hookimpl, run_cmdline
+from mailadm.db import DB
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+
+
+class AdmBot:
+    def __init__(self):
+        self.db = DB(os.getenv("MAILADM_DB"))
```
[the remark this answers is not in the document]
Comment (missytake) [reply src/mailadm/bot.py]: like this? I don't really get why we pass argv to the bot in main(). There are additional CLI arguments, right? Shouldn't they be defined somewhere?

File: src/mailadm/bot.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        else:
+            # unconditionally accept the chat
+            command.create_chat()
+            addr = command.get_sender_contact().addr
+            if command.is_system_message():
+                command.chat.send_text("echoing system message from {}:\n{}".format(addr, command))
+            else:
+                text = command.text
+                command.chat.send_text("echoing from {}:\n{}".format(addr, text))
```
[the remark this answers is not in the document]
Comment (missytake) [reply src/mailadm/bot.py]: sounds good. I also took out the weird echoing part at the end - no idea why that was still in there. probably remnants of copy-pasting the Echo bot in the beginning.

File: src/mailadm/db.py (hunk 1 above)
Comment (missytake) [reply src/mailadm/db.py]: I reverted the changes I made.

File: src/mailadm/cmdline.py [hunk 4]
```diff
@@ -46,42 +42,83 @@ def get_mailadm_db(ctx, show=False, fail_missing_config=True):
...
+    qr.add_data(chatinvite)
+    print("\nPlease scan this qr code to join a verified admin group chat:\n\n")
+    qr.print_ascii(invert=True)
+    print("\nAlternatively, copy-paste this invite to your Delta Chat desktop client:", chatinvite)
+
+    print("\nWaiting until you join the chat")
+    while chat.num_contacts() < 2:
+        time.sleep(1)
+
+    time.sleep(5)
```
[the remark this answers is not in the document]
Comment (missytake) [reply src/mailadm/cmdline.py]: I'm sending a welcome message and wait for ac_message_delivered now :)

**Comment by missytake** [comment]
Wow, I just refactored this branch (and pushed it to https://github.com/deltachat/mailadm/tree/bot-refactored), all the tests still pass, actually. We might be able to just use this, I'll take a look later if it still works.

**Comment by missytake** [comment]
Closing this in favor of #46.

**Pull request closed without merging**